tests/shared/src/test/scala/io/circe/CodecSuites.scala

```diff
@@ -75,6 +75,7 @@ class CatsCodecSuite extends CirceSuite {
 
 class CirceCodecSuite extends CirceSuite {
   checkAll("Codec[Json]", CodecTests[Json].codec)
+  checkAll("Codec[JsonObject]", CodecTests[JsonObject].codec)
```

Comment:
It's definitely a good question (see my comment above). We should test it, but writing a good `Arbitrary` instance for `JsonNumber` is non-trivial, and since the codec is extremely simple I don't think it should hold up this PR.